drivers/spi/spi_max32.c

```diff
@@ -279,13 +352,15 @@ static int spi_max32_transceive(const struct device *dev)
 
 	MXC_SPI_StartTransmission(cfg->regs);
 #else
+	key = spi_spin_lock(dev);
```

Comment:
Spin locking around the blocking transceive will prevent other interrupts. Is this the desired behavior?